### PR TITLE
Fix display issues on iOS 9.3.5

### PR DESCRIPTION
**Pull request by NovapaX**
This fixes some (bad) display issues on older devices.
ShadyCSS now correctly polyfills the default colors set in ha-style.
Before it somehow did not 'find' the css variables in the defaults because they were set with the wrong selector.

**documentation reference**
https://www.polymer-project.org/2.0/docs/upgrade#update-custom-property-syntax
see the comment in the code snippet under "After (hybrid code)" which states:
` /* In a 2.x custom-style use the html selector to set global defaults */`

**Images before and after**
img_0270
img_0271

### CODE REVIEW

**Review by andrey-git — CHANGES_REQUESTED** [review]
This needs another solution. Polymer have some default styles applied to html. When our styles are also applied to html it becomes order dependent. Thus I moved our styles to body.

**Comment by NovapaX** [comment]
It's weird because this is obviously the way it 'should' be done.
I guess #354 is solving a symptom and not the problem.

Anyway.
- turning it into `html, body {` doesn't work.
- turning the whole block into a mixin and applying it to both html and body doesn't work either.

**Comment by andrey-git** [comment]
Which browser is broken now? Safari9?

**Comment by NovapaX** [comment]
Yes. It's safari on iOS 9.
But the issue is with shadowCSS polyfill not correctly using the css variables when generating the shadowy CSS style. 
So other old browsers are probably affected with the same issue.

**Comment by andrey-git** [comment]
If you can reproduce it on some browser that works on Linux I can take back look.

**Comment by NovapaX** [comment]
Chrome 48 exhibits the exact same behavior as iOS 9.3 safari. (that's the last chrome version that does not support css-variables)

note: The safari version at hand should support css-variables, but apparently shadyCSS is not able to determine that and 'compiles' the variables into values anyway.

**Comment by andrey-git** [comment]
IIUC this is not about css variables, but about shadow DOM.

**Comment by andrey-git** [comment]
Try the following style content: https://hastebin.com/bicicayotu.lua

I basically duplicated overridable content for both `html` and `body`

**Comment by NovapaX** [comment]
Will try that. Of course that will work.
It’s about both shadow dom and CSS variables. When shadow dom is not supported but css-vars are, shadowCSS will polyfill css styles but keep the css-variable references intact.
Only when css variables are not supported it will also substitute those with the values.

**Comment by NovapaX** [comment]
You code snippet worked (of course)
It may not be elegant, but it works. 
I was however not able to reproduce the issue with the styles being overridden. (at least not in with the config panels)

**Comment by NovapaX** [comment]
Any changes still needed on this?

**Comment by NovapaX** [comment]
I've been looking at this again, and I just really think https://github.com/home-assistant/home-assistant-polymer/pull/354 needs another solution.
Unfortunately I can't easily test the zwave panel, but all configuration panels (including iframes) behave properly with global styles set on `html`.
I think we should just change it back and solve the 'order' issues a different way.

**Comment by balloob** [comment]
I think you just had the right solution in the chat. We need to import the paper styles here first so that we override them. Because it is just an ordering problem.

**Comment by NovapaX** [comment]
Polymer styles are already correctly imported at the top of ha-styles.html by importing polymer.html 
There also is no ordering issue in the default home-assistant panels as far as I can see.
It was a problem with the zwave panel (which I unfortunately cannot test easily)
see: #340

File: src/resources/ha-style.html [hunk 1]
```diff
@@ -2,7 +2,17 @@
 
 <custom-style>
 <style is="custom-style">/* remove is= on Polymer 2 */
-  body {
+/*
+  Home Assistant default styles.
+
+  In Polymer 2.0, default styles should to be set on the html selector.
+
+  To prevent overriding of thse styles by Polymer-defaults we define
```
Comment (c727) [inline src/resources/ha-style.html]: the

**Comment by andrey-git** [comment]
No, the problem a wasn't specific to zwave. I'll test again tomorrow.

**Comment by balloob** [comment]
@NovapaX the styles are not imported as part of the polymer import. That's just the framework. We should explicitly import the HTML import that is messing with the styles. Guess it's `bower_components/paper-styles/paper-styles.html` ?

**Comment by balloob** [comment]
@NovapaX did you test this and does this work?

**Comment by NovapaX** [comment]
Yes, I tested this on both chrome 48 and iOS 9 (which both behaved the same).
And of course tested it on recent chrome and safari.

note: I tested it by doing `build_frontend`, installing it into hass with `pip` and commenting out the required front-end version in `frontend/init`. So it wouldn't be in dev-mode.

**Comment by ingoha** [comment]
Should the frontend work with iOS 9.3.5 (hass version 0.79.3)? For me it does not :-/

**Pull request merged**